pkg/artifacts/signable.go

```diff
@@ -25,6 +27,14 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+const (
+	digestFormat = `sha[0-9]+\:[a-zA-Z0-9]+`
```

Comment:
I think this will match some unintended SHAs - e.g. `sha25600000000000000...:a`. You probably want something like this instead - https://regex101.com/r/NI7n8j/2 (though it doesn't check whether the checksum algo is valid). We should add tests for this regex.

I realized after the last round that you were likely basing the old implementation off of go-containerregistry. You might be interested in https://github.com/google/go-containerregistry/pull/1395 which updates that implementation to use github.com/opencontainers/go-digest as an alternative.